binrw/src/builder_types.rs

```diff
@@ -29,18 +29,34 @@ pub fn builder_helper<T: BinrwNamedArgs>(_: PhantomData<T>) -> T::Builder {
 }
 
 /// A macro for creating a binrw argument type
+///
+/// This macro avoids taking an explicit type by inferring the type it should create.
+/// In general, the result should have an explicit type *immediately*.
+/// i.e. being passed to a function or let binding with specific type.
 #[macro_export]
 macro_rules! args {
     (@ifn { $value:expr } $name:ident) => { $value };
     (@ifn {} $name:ident) => { $name };
     ($($name:ident $(: $value:expr)?),*) => {
         {
+            // I'll use Ret to represent the type of the block
```

Comment:
```suggestion
            // I'll use Ret to represent the type of the block.
```
This was hard to parse mentally since implicitly, comments wrap to the next line and "block token" is a valid phrase.